VITCMUN/munverse
# Dockerizing the application and update CircleCI pipeline

Reference #54, #56

Changes made:
- Move database logic to db/db.js from app.js for readability
- Populate event and council using environment variables
- Create a default admin user using environment variables
- Use connect-mongo for session storage instead of plain express-session for performance reasons
- Dockerfiles
   - Dockerfile: for development purpose with live reloading (nodemon) and carbon image
   - build.Dockerfile: for production purpose with an alpine image
- Docker Compose
  - docker-compose.yaml: production config
  - docker-compose.dev.yaml: dev config with bind volume and mongo UI
- Scripts
  - startup.sh: run appropriate server for the env
  - wait-for-it.sh: wait for mongo to start before starting the web app
- CircleCI now has a two-step workflow.
    - test the code
    - build a docker image and push to dockerhub (https://hub.docker.com/r/vitcmun/munverse)
- Starting and Stopping development container commands

This is a huge PR with many changes. I was planning to keep it small but had to update things here and there to make it compatible with the changes I made. In short, the code is dockerized now..

Future work to consider
- Add a script to automatically add users from a CSV
- CircleCI's build tag should be taken from package.json rather than hardcoded value

Also, the media folder needs to be sorted. The media folder should not exist in a docker but should be served using a CDN server. Packaging it with the code is a bad thing. @kunal768, let's open a discord discussion on this issue to decide further.

I also made the required changes to the Wiki :tada:
